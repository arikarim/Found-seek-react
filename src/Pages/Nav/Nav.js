import React from "react";
import { Link, useHistory } from "react-router-dom";
import { logout } from "../../PureFunctions/Logout";
import "./Nav.css";
const Nav = ({user, setUser }) => {
  const history = useHistory();

  const toke = JSON.parse(localStorage.getItem("token"));
  return (
    <nav className="d-flex py-2 flex-column flex-md-row justify-content-between">
      <h3 className="my-auto mx-auto mx-md-2">Found&Seek</h3>
      {toke === "" || user === "Not Logged" ? (
        <ul className="list-unstyled my-auto d-flex py-3 justify-content-center justify-content-md-end">
          <li className="mx-3">
            <Link className="text-decoration-none link-light" to="/">
              Home
            </Link>
          </li>
          <li className="mx-3">
            <Link className="text-decoration-none link-light" to="/signup">
              Sign Up
            </Link>
          </li>
          <li className="mx-3">
            <Link className="text-decoration-none link-light" to="/login">
              Login
            </Link>
          </li>
        </ul>
      ) : (
        <ul className="list-unstyled my-auto d-flex py-3 justify-content-center justify-content-md-end">
          <li className="mx-3">
            <Link className="text-decoration-none link-light" to="/">
              Home
            </Link>
          </li>
          <li className="mx-3">
            <Link className="text-decoration-none link-light" to="/my_devices">
              My devices
            </Link>
          </li>
          <li className="mx-3">
            <Link
              className="text-decoration-none link-light"
              to="/create_device"
            >
              Create device
            </Link>
          </li>
          <li className="mx-3">
            <button
              className="logout-btn"
              type="submit"
              onClick={(e) => logout(setUser, history)}
            >
              Log out
            </button>
          </li>
        </ul>
      )}
    </nav>
  );
};

export default Nav;
