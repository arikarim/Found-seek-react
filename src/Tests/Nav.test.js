import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Nav from "../Pages/Nav";

describe("Header with Link", () => {
  test("should match with snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
