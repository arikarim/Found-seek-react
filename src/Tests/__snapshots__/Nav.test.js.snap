// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<nav
  className="d-flex py-2 flex-column flex-md-row justify-content-between"
>
  <h3
    className="my-auto mx-auto mx-md-2"
  >
    Found&Seek
  </h3>
  <ul
    className="list-unstyled my-auto d-flex py-3 justify-content-center justify-content-md-end"
  >
    <li
      className="mx-3"
    >
      <a
        className="text-decoration-none link-light"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="mx-3"
    >
      <a
        className="text-decoration-none link-light"
        href="/my_devices"
        onClick={[Function]}
      >
        My devices
      </a>
    </li>
    <li
      className="mx-3"
    >
      <a
        className="text-decoration-none link-light"
        href="/create_device"
        onClick={[Function]}
      >
        Create device
      </a>
    </li>
    <li
      className="mx-3"
    >
      <button
        className="logout-btn"
        onClick={[Function]}
        type="submit"
      >
        Log out
      </button>
    </li>
  </ul>
</nav>
`;
